static/src/js/actions/toastNotifications.js

```diff
@@ -0,0 +1,211 @@
+  topCenter: 'top-center',
+  topRight: 'top-right'
+}
+
+const notificationDefaults = {
+  type: ToastType.info,
+  autoClose: 15000,
+  position: ToastPosition.topRight,
+  hideProgressBar: false,
+  newestOnTop: true,
+  closeOnClick: true,
+  closeButton: true,
+  rtl: false,
+  pauseOnVisibilityChange: true,
+  draggable: false,
+  pauseOnHover: true,
+  style: {
+    color: '#fcfcfc'
+  },
+  progressStyle: {
+    background: 'navy'
+  }
+}
+
+// Sets the default notification settings
+toast.configure({
+  ...notificationDefaults,
+  bodyClassName: 'root'
+})
+
+export const pushingNotification = notification => ({
+  type: PUSHING_NOTIFICATION,
+  notification
+})
+
+export const notificationPushed = id => ({
+  type: NOTIFICATION_PUSHED,
+  id
+})
+
+export const notificationDismissed = id => ({
+  type: NOTIFICATION_DIMISSED,
+  id
+})
+
+/**
+ * @name addNotification
+ * @param {object} param0 Notification object.
+ * @description Call toast system to render the notification.
+ */
+const addNotification = ({
+  content,
+  type,
+  autoClose,
+  position,
+  onClose
+}) => {
+  toast[type || notificationDefaults.type](content, {
+    autoClose,
+    position,
+    onClose
+  })
+}
+
+/**
+ * @name pushSuccessNotification
+ * @param {React.Node} content
+ * @description Generates a default success notification with
+ * the content value inside the toast container.
+ */
+export const pushSuccessNotification = content => (dispatch) => {
+  const id = uuid()
+  const notification = {
+    id,
+    content,
+    type: ToastType.success,
+    pushed: false,
+    dismissed: false,
+    onClose: () => dispatch(notificationDismissed(id)),
+    time: new Date()
+  }
+  dispatch(pushingNotification(notification))
+  addNotification(notification)
+  dispatch(notificationPushed(id))
+}
+
+/**
+ * @name pushWarningNotification
+ * @param {React.Node} content
+ * @description Generates a default warning notification with
+ * the content value inside the toast container.
+ */
+export const pushWarningNotification = content => (dispatch) => {
+  const id = uuid()
+  const notification = {
+    id,
+    content,
+    type: ToastType.warning,
+    pushed: false,
+    dismissed: false,
+    onClose: () => dispatch(notificationDismissed(id)),
+    time: new Date()
+  }
+  dispatch(pushingNotification(notification))
+  addNotification(notification)
+  dispatch(notificationPushed(id))
+}
+
+/**
+ * @name pushErrorNotification
+ * @param {React.Node} content
+ * @description Generates a default error notification with
+ * the content value inside the toast container.
+ */
+export const pushErrorNotification = content => (dispatch) => {
+  const id = uuid()
+  const notification = {
+    id,
+    content,
+    type: ToastType.error,
+    autoClose: false,
+    pushed: false,
+    dismissed: false,
+    onClose: () => dispatch(notificationDismissed(id)),
+    time: new Date()
+  }
+  dispatch(pushingNotification(notification))
+  addNotification(notification)
```

Comment:
Is there a reason that `dispatch` is used for some of the actions but not others?

Reply:
https://github.com/nasa/earthdata-search/pull/1245#discussion_r501121788